reference/configuration/doctrine.rst

```diff
@@ -499,4 +499,150 @@ can be placed directly under ``doctrine.orm`` config level.
+
+For example, the following looks for entity classes in the ``App\Entity`` namespace in the
+`src/Entity` directory and gives them an ``App`` alias (so you can say things like `App:Post`):
+
+.. configuration-block::
+
+    .. code-block:: yaml
+    
+        doctrine:
+                # ...
+                orm:
+                    # ...
+                    mappings:
+                        # ...
+                        SomeEntityNamespace:
+                            type: annotation
+                            dir: %kernel.root_dir%/../src/Entity
+                            is_bundle: false
+                            prefix: App\Entity
+                            alias: App
+                            
+    .. code-block:: xml
+    
+        <?xml version="1.0" charset="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:doctrine="http://symfony.com/schema/dic/doctrine">
+        
+            <doctrine:config>
+                <doctrine:orm>
+                    <mapping name="SomeEntityNamespace" 
+                        type="annotation"
+                        dir="%kernel.root_dir%/../src/Entity"
+                        is-bundle="false"
+                        prefix="App\Entity"
+                        alias="App"
+                    />
+                </doctrine:orm>
+            </doctrine:config>
+        </container>
+        
+    .. code-block:: php
+            
+        $container->loadFromExtension('doctrine', array(
+            'orm' => array(
+                'auto_mapping' => true,
+                'mappings' => array(
+                    'SomeEntityNamespace' => array(
+                        'type'      => 'annotation',
+                        'dir'       => '%kernel.root_dir%/../src/Entity',
+                        'is_bundle' => false,
+                        'prefix'    => 'App\Entity',
+                        'alias'     => 'App',
+                    ),
+                ),
+            ),
+        ));
+
+Detecting a Mapping Configuration Format
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the ``type`` on the bundle configuration isn't set,
+the DoctrineBundle will try to detect the correct mapping configuration format for
+the bundle. The DoctrineBundle will look for files following the filename pattern 
+``[BundleDirectory]/[ResourceConfigurationDirectory]/*.orm.[Extension]``.
+The ``ResourceConfigurationDirectory``, by default, points to
+``Resources/config/doctrine``.
+
+The bundle looks for (in this order) XML, YAML and PHP files.
+Using the ``auto_mapping`` feature, every bundle can have only one configuration format.
+The bundle will stop as soon as it locates one.
+
+If it wasn't possible to determine a configuration format, it will check if
+there is an ``Entity`` folder in the bundle's root directory.
+If the folder does not exist, Doctrine will fall back to using an annotation driver.
```

Comment:
I think this is still written just for "bundles" - I think it should be more generic. The only difference between bundle mapping and non-bundle mapping (I think) is that the ``dir`` for a bundle mapping
will be relative to the bundle's dir, whereas for non-bundles, it's the full, absolute path.